### DISCUSSION TITLE
Better docs url scheme

### DISCUSSION THREAD

**Discussion by acalvino4** [craftcms/docs#356] [Ideas]
Right now each version of the docs is hosted at a URL of the form https://craftcms.com/docs/3.x/xxxxxx...
I want to propose that the latest version is always hosted at https://craftcms.com/docs/xxxxxxx (dropping the version number).
The main reason is so that when a new version comes out, it gets all the seo juice from previous versions of the docs.

One example of a project that does this (though far from the only one) is tailwind - they just recently released version 3 and bumped all version 2 docs to https://v2.tailwindcss.com/docs.

One example of another project that does not do this is twig, and it is annoying that I frequently get Twig 2 docs as the most prominent in search results, despite the fact that version 3 has been out for a long time. I don't want this to be the case for Craft 4.

The only real hiccup I am aware of for this is that a few doc pages from the old version may not exist for the new version. You would have to make sure these are redirected appropriately.

**Comment by acalvino4** [comment]
@mattstein Don't know if anyone has looked at this post, but now that Craft 4 has been released, I just want to point out that _every_ time I search, the top results are always for the Craft 3 docs, in other words, what I described above is exactly what happened. It would be awesome if we could fix this!

**Reply by mattstein** [reply]
I’m reluctant to change the URL scheme because we arrived at it to solve a few problems, including navigation between product versions (even when page URLs change) and directing people to the most recent version when they land on the docs from search engines.

Sub-optimal SEO for the latest version has come up before though, which is why we finally added proper canonical tags [recently](https://github.com/craftcms/docs/blob/main/CHANGELOG.md#2022-06-24)—similarly to how Laravel’s docs are managed. It may be too soon to measure the impact of that change, but we can revisit if things don’t improve in the near future.

**Reply by acalvino4** [reply]
Fair enough! I guess we'll hopefully see an improvement over next few months.

**Reply by acalvino4** [reply]
I'm still consistently getting Craft 3 docs in search results.

Did a quick peek at the page source code, and I don't see canonical tags (https://craftcms.com/docs/commerce/3.x/tax.html):
![image](https://user-images.githubusercontent.com/43254861/186230618-a30d0fda-d0cf-4be1-844b-2db8d08a137f.png)

I also checked the response headers and nothing there either. I'm far from an SEO pro, so I'm not sure if there's another way of setting that, but I wonder if that's why google isn't updating the results. I also don't know how long google's crawler normally takes to update this kind of thing, so while my intuition is that it should have picked it up by now if it was set up correctly, I understand that we may just need to wait longer.

**Reply by acalvino4** [reply]
a few months later, still getting v3 docs as primary search results: 
![image](https://user-images.githubusercontent.com/43254861/200396614-3cc486c4-1106-4c42-80b9-9fc4c2dcc878.png)

**Comment by AugustMiller** [comment]
Thanks for bumping this—something is definitely not right with how Vuepress is handling additional `<head>` content, and our canonical URLs are caught up in that.

I'll be looking at it in greater depth this week!
